src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,22 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be more than 6 characters");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Login cannot repeat");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException("Login must be more than 6 characters");
```

Comment:
The exception message here is incorrect. It should state 'Password must be more than 6 characters' instead of 'Login must be more than 6 characters'. This is an error related to the task requirements, as the message should accurately reflect the validation being performed.